FreeRTOS-Plus/Demo/coreSNTP_Windows_Simulator/DemoTasks/SNTPClientTask.c

```diff
@@ -0,0 +1,723 @@
+ * @param[in] bytesToSend The size of data in @p pBuffer to send.
+ *
+ * @return Returns the return code of FreeRTOS UDP send API, FreeRTOS_sendto, which returns
+ * 0 for error or timeout OR the number of bytes sent over the network.
+ */
+static int32_t UdpTransport_Send( NetworkContext_t * pNetworkContext,
+                                  uint32_t serverAddr,
+                                  uint16_t serverPort,
+                                  const void * pBuffer,
+                                  size_t bytesToSend );
+
+/**
+ * @brief The demo implementation of the @ref UdpTransportRecvFrom_t function
+ * of the UDP transport interface to allow the coreSNTP library to perform
+ * network operation of reading expected time response over UDP from
+ * provided time server.
+ *
+ * @param[in] pNetworkContext This will be the NetworkContext_t context object
+ * representing the FreeRTOS UDP socket to use for network send operation.
+ * @param[in] pTimeServer The IPv4 address of the time server to receive data from.
+ * @param[in] serverPort The port of the server to receive data from.
+ * @param[out] pBuffer The demo-supplied network buffer of size, SNTP_CONTEXT_NETWORK_BUFFER_SIZE,
+ * that will be filled with data received from the network.
+ * @param[in] bytesToRecv The expected number of bytes to receive from the network
+ * for the server response server.
+ *
+ * @return Returns one of the following:
+ * - 0 for timeout in receiving any data from the network (by translating the
+ * -pdFREERTOS_ERRNO_EWOULDBLOCK return code from FreeRTOS_recvfrom API )
+ *                         OR
+ * - The number of bytes read from the network.
+ */
+static int32_t UdpTransport_Recv( NetworkContext_t * pNetworkContext,
+                                  uint32_t serverAddr,
+                                  uint16_t serverPort,
+                                  void * pBuffer,
+                                  size_t bytesToRecv );
+
+/**
+ * @brief The demo implementation of the @ref SntpGetTime_t interface
+ * for obtaining system clock time for the coreSNTP library.
+ *
+ * @param[out] pTime This will be populated with the current time from
+ * the system.
+ */
+static void sntpClient_GetTime( SntpTimestamp_t * pCurrentTime );
+
+/**
+ * @brief The demo implementation of the @ref SntpSetTime_t interface
+ * for correcting the system clock time based on the  time received
+ * from the server response and the clock-offset value calculated by
+ * the coreSNTP library.
+ *
+ * @note This demo uses either the "slew" OR "step" methodology of system
+ * clock correction based on the use-case:
+ * 1. "Step" correction is used if:
+ *   - System time is ahead of server time so that system time is immediately
+ *     corrected instead of potentially receding back in time with a "slew"
+ *     correction approach when the system time is ahead by a large margin in the
+ *     order of days, months or more.
```

Comment:
```suggestion
 *   - System time is ahead of server time so that system time is immediately
 *     corrected instead of potentially receding back in time with a "slew"
 *     correction approach.
```